tls/s2n_connection.h

```diff
@@ -148,6 +148,9 @@ struct s2n_connection {
      */
     uint16_t max_outgoing_fragment_length;
 
+    /* Negotiated TLS extension Maximum fragment size */
+    uint16_t max_fragment_length;
```

Comment:
A field for `max_outgoing_fragment_length` and `max_fragment_length` is a little confusing. How about two separate fields for `max_outgoing_fragment_length` and `max_incoming_fragment_length`? These fields can be governed by the max_frag extension and the `s2n_prefer_*` api calls as needed.